debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/VirtualThreadsCommand.java

```diff
@@ -0,0 +1,110 @@
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+package com.ibm.j9ddr.vm29.tools.ddrinteractive;
+
+import java.io.PrintStream;
+
+import com.ibm.j9ddr.CorruptDataException;
+import com.ibm.j9ddr.tools.ddrinteractive.Command;
+import com.ibm.j9ddr.tools.ddrinteractive.Context;
+import com.ibm.j9ddr.tools.ddrinteractive.DDRInteractiveCommandException;
+import com.ibm.j9ddr.vm29.j9.DataType;
+import com.ibm.j9ddr.vm29.j9.J9ObjectFieldOffset;
+import com.ibm.j9ddr.vm29.pointer.ObjectReferencePointer;
+import com.ibm.j9ddr.vm29.pointer.PointerPointer;
+import com.ibm.j9ddr.vm29.pointer.generated.J9JavaVMPointer;
+import com.ibm.j9ddr.vm29.pointer.generated.J9ObjectPointer;
+import com.ibm.j9ddr.vm29.pointer.helper.J9ObjectHelper;
+import com.ibm.j9ddr.vm29.pointer.helper.J9RASHelper;
+import com.ibm.j9ddr.vm29.types.UDATA;
+
+/**
+ * vthreads command lists all the virtual threads.
+ *
+ * Example:
+ *     !vthreads
+ * Example output:
+ *     !continuationstack 0x00007fe78c0f9600 !j9vmcontinuation 0x00007fe78c0f9600 !j9object 0x0000000706401588 (Continuation) !j9object 0x0000000706400FB0 (VThread) - name1
+ *     !continuationstack 0x00007fe78c23aa80 !j9vmcontinuation 0x00007fe78c23aa80 !j9object 0x0000000706424F90 (Continuation) !j9object 0x0000000706424EF0 (VThread) - name2
+ *     !continuationstack 0x00007fe78c244ac0 !j9vmcontinuation 0x00007fe78c244ac0 !j9object 0x00000007064250D8 (Continuation) !j9object 0x0000000706425038 (VThread) - name3
+ *     ...
+ */
+public class VirtualThreadsCommand extends Command {
+	public VirtualThreadsCommand() {
+		addCommand("vthreads", "", "Lists virtual threads");
+	}
+
+	@Override
+	public void run(String command, String[] args, Context context, PrintStream out) throws DDRInteractiveCommandException {
+		try {
+			J9JavaVMPointer vm = J9RASHelper.getVM(DataType.getJ9RASPointer());
+			if (JavaVersionHelper.ensureMinimumJavaVersion(19, vm, out)) {
+				displayVirtualThreads(vm, out);
+			}
+		} catch (CorruptDataException | NoSuchFieldException e) {
+			throw new DDRInteractiveCommandException(e);
+		}
+	}
+
+	/**
+	 * Prints all the live virtual threads
+	 *
+	 * @param vm the J9JavaVMPointer of the virtual machine.
+	 * @param out the PrintStream to write output to.
+	 */
+	private static void displayVirtualThreads(J9JavaVMPointer vm, PrintStream out) throws CorruptDataException, NoSuchFieldException {
+		PointerPointer mainVirtualThread = vm.liveVirtualThreadList();
+
+		UDATA linkNextOffset = vm.virtualThreadLinkNextOffset();
```

Comment:
Suggest you reorder some computations to avoid duplication:
```
		if (mainVirtualThread.isNull()) {
			return;
		}

		J9ObjectPointer root = J9ObjectPointer.cast(mainVirtualThread.at(0));

		if (root.isNull()) {
			return;
		}
```